src/neo/Ledger/MemoryPool.cs

```diff
@@ -361,6 +361,11 @@ internal void UpdatePoolForBlockPersisted(Block block, DataCache snapshot)
                 _txRwLock.ExitWriteLock();
             }
 
+            // If we know about headers of future blocks, no point in verifying transactions from the unverified tx pool
+            // until we get caught up.
+            if (block.Index > 0 && Blockchain.Singleton.HeaderCache.Count > 0)
```

Comment:
we should check the last index instead of the count, isn't it?

Reply:
They are equivalent.

Reply:
But we have a limit of 10K headers

Reply:
What do you mean?

Reply:
`Blockchain.Singleton.HeaderCache.Count > 0` is equivalent to `height < headerHeight`.

Reply:
`block.Index > 0` it's required?

Reply:
It's not required.

Reply:
We need it. Because when `block.Index == 0`, `Blockchain.Singleton` is not available.